src/components/sidebar/SidebarMenu.tsx

```diff
@@ -1,42 +1,43 @@
-import { Link } from "react-router-dom";
-import { Sidebar, Menu, MenuItem, SubMenu } from "react-pro-sidebar";
-import MenuRoundedIcon from "@mui/icons-material/MenuRounded";
+import AccountCircleRoundedIcon from "@mui/icons-material/AccountCircleRounded";
 import GridViewRoundedIcon from "@mui/icons-material/GridViewRounded";
+import LogoutRoundedIcon from "@mui/icons-material/LogoutRounded";
+import MenuRoundedIcon from "@mui/icons-material/MenuRounded";
 import ReceiptRoundedIcon from "@mui/icons-material/ReceiptRounded";
 import SettingsApplicationsRoundedIcon from "@mui/icons-material/SettingsApplicationsRounded";
-import AccountCircleRoundedIcon from "@mui/icons-material/AccountCircleRounded";
-import LogoutRoundedIcon from "@mui/icons-material/LogoutRounded";
+import { Menu, MenuItem, Sidebar, SubMenu } from "react-pro-sidebar";
+import { Link } from "react-router-dom";
 
 export const SidebarMenu = ({ onClose }: any) => {
   const menuItems = [
     {
+      to: "",
       className: "menu1",
       icon: <MenuRoundedIcon onClick={onClose} />,
       label: <h2>Your Box</h2>,
     },
     {
       to: "/dashboard",
-      icon: <GridViewRoundedIcon />,
+      icon: <GridViewRoundedIcon onClick={onClose} />,
       label: "Dashboard",
     },
     {
       to: "/invoices",
-      icon: <ReceiptRoundedIcon />,
+      icon: <ReceiptRoundedIcon onClick={onClose} />,
       label: "Invoices",
     },
     {
       label: "Settings",
-      icon: <SettingsApplicationsRoundedIcon />,
+      icon: <SettingsApplicationsRoundedIcon onClick={onClose} />,
       subMenu: [
         {
-          icon: <AccountCircleRoundedIcon />,
+          icon: <AccountCircleRoundedIcon onClick={onClose} />,
           label: "Account",
         },
       ],
     },
     {
       to: "/login",
-      icon: <LogoutRoundedIcon />,
+      icon: <LogoutRoundedIcon onClick={onClose} />,
```

Comment:
Poniżej masz menuItems map i przypisujesz bo SubMenu index (nie moge dodac komentarza w niezmienionym kodzie chyba)

Tutaj artykuł dlaczego index jako key jest złym wyborem:
https://medium.com/react-in-the-real-world/why-you-should-never-ever-use-indices-as-react-keys-8329472170be